spec.emu

```diff
@@ -43,15 +43,16 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
     <emu-clause id="sec-placeholder"><h1>Placeholder to ensure correct clause numbering</h1></emu-clause>
     <emu-clause id="sec-placeholder"><h1>Placeholder to ensure correct clause numbering</h1></emu-clause>
 
-    <emu-clause id="sec-createnonenumerabledatapropertyorthrow">
-      <h1><ins>CreateNonEnumerableDataPropertyOrThrow ( _O_, _P_, _V_ )</ins></h1>
-      <p><ins>The abstract operation CreateNonEnumerableDataPropertyOrThrow takes arguments _O_ (an Object), _P_ (a Property Key), and _V_ (an ECMAScript language value). It is used to create a new non-enumerable own property of an object. It throws a *TypeError* exception if the requested property update cannot be performed. It performs the following steps when called:</ins></p>
+    <emu-clause id="sec-createnonenumerabledataproperty">
+      <h1><ins>CreateNonEnumerableDataProperty ( _O_, _P_, _V_ )</ins></h1>
+      <p><ins>The abstract operation CreateNonEnumerableDataProperty takes arguments _O_ (an Object), _P_ (a property key), and _V_ (an ECMAScript language value). It performs the following steps when called:</ins></p>
       <emu-alg>
+        1. <ins>Assert: Type(_O_) is Object.</ins>
+        1. <ins>Assert: IsPropertyKey(_P_) is true.</ins>
         1. <ins>Let _newDesc_ be the PropertyDescriptor { [[Value]]: _V_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
-        1. <ins>Return ? DefinePropertyOrThrow(_O_, _P_, _newDesc_).</ins>
+        1. <ins>Return ! DefinePropertyOrThrow(_O_, _P_, _newDesc_).</ins>
```

Comment:
in what way is this infallible?? It could be called with an O where P is already nonconfigurable.

Reply:
This call cannot fail in this context because _O_ is the result of OrdinaryCreateFromConstructor and has not yet been exposed to user code. I believe that's also why the call is prefixed with "!" in 262 today.

Reply:
The call site can’t throw. This abstract operation can’t make that assumption.